Detected branch is `undefined` in GH Action on pull_request

I'm working on integrating `gh-openapi-docs` to automatically build and publish the openapi specifications and docs. When pushing to the branches `develop` and `master`, I want to build and publish. When pushing a commit to a branch used by a PR, I only want to build.

Here is my current job (WIP):

```
  build-and-publish:
    needs: [lint]
    runs-on: ubuntu-latest
    strategy:
      matrix:
        api:
        - "data-node"
        # - "date-annotator"
    steps:
    - uses: actions/checkout@v2
    - name: Set variables
      id: vars
      run: |
        echo ::set-output name=version::test
        echo ::set-output name=publish::false
        if [[ ${GITHUB_REF} == refs/tags/* ]]
        then
          echo ::set-output name=version::${GITHUB_REF#refs/tags/}
          echo ::set-output name=publish::true
        elif [[ ${GITHUB_REF} == refs/heads/main ]]
        then
          echo ::set-output name=version::
          echo ::set-output name=publish::true
        elif [[ ${GITHUB_REF} == refs/heads/develop ]]
        then
          echo ::set-output name=version::develop
          echo ::set-output name=publish::true
        fi
    - name: Use Node.js ${{ env.node_version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ env.node_version }}
    - run: npm install -g @ga4gh/gh-openapi-docs
    - name: Create .spec-docs.json
      run: |
        printf "%s\n" \
        '{' \
        '"apiSpecPath": "openapi/${{ steps.vars.outputs.version }}/openapi.yaml",' \
        '"docsRoot": "docs",' \
        '"defaultBranch": "",' \
        '"branchPathBase": "preview"' \
        '}' | tee -a .spec-docs.json
    - name: Build openapi specifications and docs
      run: gh-openapi-docs
    - run: ls -al
```

Here is the output and error thrown by `gh-openapi-docs` when triggered by a PR. The error is because the branch is identified as `undefined`.

```
{
  sha: 'f49b5c5e9f67ef4fbb7170d798d3645f04ccaae4',
  abbreviatedSha: 'f49b5c5e9f',
  branch: undefined,
  tag: null,
  committer: 'GitHub <noreply@github.com>',
  committerDate: '2020-08-29T16:25:22.000Z',
  author: 'Thomas Schaffter <thomas.schaffter@gmail.com>',
  authorDate: '2020-08-29T16:25:22.000Z',
  commitMessage: 'Merge b3ed610a6c073dbf251269fe843886166e0afc33 into ab775570d96d69bf8b2909bd7b68a906fcf26f36',
  root: '/home/runner/work/nlp-sandbox-schemas/nlp-sandbox-schemas',
  commonGitDir: '/home/runner/work/nlp-sandbox-schemas/nlp-sandbox-schemas/.git',
  worktreeGitDir: '/home/runner/work/nlp-sandbox-schemas/nlp-sandbox-schemas/.git',
  lastTag: null,
  commitsSinceLastTag: Infinity,
  parents: [ 'b3ed610a6c073dbf251269fe843886166e0afc33' ]
}
webpack:///./src/lib/config.js?:33
    return path__WEBPACK_IMPORTED_MODULE_1___default.a.join(root, branchPathBase, currentBranch.toLowerCase());
                                                                                                ^

TypeError: Cannot read property 'toLowerCase' of undefined
    at constructBranchPath (webpack:///./src/lib/config.js?:33:97)
    at eval (webpack:///./src/lib/config.js?:38:15)
    at Module../src/lib/config.js (/opt/hostedtoolcache/node/12.18.3/x64/lib/node_modules/@ga4gh/gh-openapi-docs/dist/bundle.js:4409:1)
    at __webpack_require__ (/opt/hostedtoolcache/node/12.18.3/x64/lib/node_modules/@ga4gh/gh-openapi-docs/dist/bundle.js:21:30)
    at eval (webpack:///./src/lib/tasks.js?:5:65)
    at Module../src/lib/tasks.js (/opt/hostedtoolcache/node/12.18.3/x64/lib/node_modules/@ga4gh/gh-openapi-docs/dist/bundle.js:4505:1)
    at __webpack_require__ (/opt/hostedtoolcache/node/12.18.3/x64/lib/node_modules/@ga4gh/gh-openapi-docs/dist/bundle.js:21:30)
    at eval (webpack:///./src/lib/index.js?:3:64)
    at Module../src/lib/index.js (/opt/hostedtoolcache/node/12.18.3/x64/lib/node_modules/@ga4gh/gh-openapi-docs/dist/bundle.js:4457:1)
    at __webpack_require__ (/opt/hostedtoolcache/node/12.18.3/x64/lib/node_modules/@ga4gh/gh-openapi-docs/dist/bundle.js:21:30)
##[error]Process completed with exit code 1.
```

A solution could be to make `gh-openapi-docs` initialize the branch name as the *string* "undefined" and then if the value of the detected branch name is different from the *value* `undefined`, set the value to the detected branch name. This way the branch name in `gh-openapi-docs` will always be set to a string and the above error should be solved.

--- a/src/gh-openapi-docs.js
+++ b/src/gh-openapi-docs.js
@@ -1,7 +1,7 @@
 import '@babel/polyfill';
 // const updater = require('update-notifier');
 import parseArgs from 'yargs-parser';
-import pkg from '../package.json';
+// import pkg from '../package.json';
 import release from './lib';
 
 const aliases = {

--- a/src/lib/config.js
+++ b/src/lib/config.js
@@ -1,4 +1,4 @@
-import _ from 'lodash';
+import { merge } from 'lodash';
 import path from 'path';
 import envConfig from './environment'
 import fs from 'fs';
@@ -37,7 +37,7 @@ const deployConfig = {
 };
 // Export the config object based on the NODE_ENV
 // ==============================================
-const config = _.merge(
+const config = merge(
   localConfig,
   envConfig,
   deployConfig

--- a/src/lib/environment.js
+++ b/src/lib/environment.js
@@ -1,14 +1,14 @@
-import _ from 'lodash';
+import { merge } from 'lodash';
 import origin from 'remote-origin-url';
 import getRepoInfo from 'git-repo-info';
 /*eslint no-process-env:0*/
 
 var repoInfo = getRepoInfo();
-repoInfo.branch = repoInfo.branch || process.env.TRAVIS_BRANCH;
+repoInfo.branch = repoInfo.branch || process.env.TRAVIS_BRANCH || "undefined";
 
 const env = process.env.NODE_ENV;
 const repoOrigin = origin.sync();
-const environment = _.merge(repoInfo, {
+const environment = merge(repoInfo, {
     env,
     repoOrigin
 });

--- a/src/lib/tasks.js
+++ b/src/lib/tasks.js
@@ -1,4 +1,3 @@
-import _ from 'lodash';
 import Logger from './log';
 import Config from './config';
 import { fetchPages } from './gh-pages';

